### PR TITLE
TS-26925: Enable SSL validation by default

### PR DESCRIPTION
**Pull request by heinemann-cqse**
Addresses issue [TS-26925](https://cqse.atlassian.net/browse/TS-26925)

- [ ] Changes are tested adequately
- [x] Agent's README.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [ ] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566)
- [ ] [TGA Tutorial](https://docs.teamscale.com/tutorial/setting-up-tga-java) updated
- [ ] [TIA Tutorial](https://docs.teamscale.com/tutorial/tia-java/) updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

**Review by DreierF — CHANGES_REQUESTED** [review]
Please adjust the documentation (README.md) regarding the `validate-ssl` option and add an entry to the `CHANGELOG.md` file as requested in the PR template

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ We use [semantic versioning](http://semver.org/):
 
 # Next Release
 - [feature] Option `teamscale-revision-manifest-jar` which can be used to read the git commit revision from the `META-INF/MANIFEST.MF` in the Jar file. The field must be named `Revision` and must be provided as a main attribute. Alternatively the field can be called `Git_Commit` and must be an attribute of an entry called `Git`.
+- [fix] _teamscale-client_: SSL validation is now enabled by default
```
Comment (DreierF) [inline CHANGELOG.md:9]: This should be `[breaking change]` as it cannot be release as a patch update and also does not only hold for teamscale client but also for the jacoco agent and the teamscale gradle plugin

**Review by DreierF — CHANGES_REQUESTED** [review]

**Review by DreierF — APPROVED** [review]

**Comment by karottenreibe** [comment]
@DreierF @heinemann-cqse please document for users how, if ssl validation is enabled, they can configure their custom certificates. We have lots of customers with their own certificate authority or self-signed certificates. They need to be able to get the agent running with that without a hassle.

I myself have no idea how to actually do that. Will the normal Java `-D` flags work? Please test this and document the process. I need to be able to tell the next customer what to do in that case.

For testing, this might help: https://badssl.com/

**Comment by karottenreibe** [comment]
Created TS-28539 for this

**Pull request merged**